examples/kubeflow/mnist/pipeline.py

```diff
@@ -82,21 +88,124 @@ def kubeflow_pipeline(s3_endpoint: str, input_bucket: str):
 
     return op(s3_endpoint, input_bucket)
 
-def main():
-    client = kfp.Client()
-
-    run_id = client.create_run_from_pipeline_func(
-        kubeflow_pipeline,
-        arguments={
-            "s3_endpoint": os.environ["S3_ENDPOINT"],
-            "input_bucket": "input",
-        }
-    ).run_id
 
-    print("run id: {}".format(run_id))
-
-    j = client.wait_for_run_completion(run_id, 60)
-    assert j.run.status == 'Succeeded'
+def pipeline_id(client: kfp.Client, name: str):
+    """Gets the ID of the kubeflow pipeline with the name 'name'
+    Args:
+      name of the pipeline
+    Returns:
+      id of the pipeline
+    """
+    page_token = ""
+    while True:
+        p = client.list_pipelines(page_token=page_token, page_size=100)
+        if p.pipelines is None:
+            return ""
+        for p in p.pipelines:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
+
+def experiment_id(client: kfp.Client, name: str):
+    """Gets the ID of the kubeflow experiment with the name 'name'
+    Args:
+      name of the experiment
+    Returns:
+      id of the experiment
+    """
+    page_token = ""
+    while True:
+        p = client.list_experiments(page_token=page_token, page_size=100)
+        for p in p.experiments:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
```

Comment:
similar simplification could be made on this loop

Reply:
Done